drivers/scheduler/k8s/k8s.go

```diff
@@ -1482,51 +1503,45 @@ func (k *k8s) StartSchedOnNode(n node.Node) error {
 	return nil
 }
 
-// CreateCRDObjects and Validate their deployment
-func (k *k8s) CreateCRDObjects(ctx *scheduler.Context, timeout, retryInterval time.Duration) error {
+// createCRDObjects and Validate their deployment
+func (k *k8s) createCRDObjects(specObj interface{}, timeout, retryInterval time.Duration) (interface{}, error) {
 	var err error
+	var obj interface{}
 	k8sOps := k8s_ops.Instance()
-	for _, specObj := range ctx.App.SpecList {
-		if obj, ok := specObj.(*stork_api.ClusterPair); ok {
-			logrus.Info("Applying clusterpair spec")
-			err = k8sOps.CreateClusterPair(obj)
-			if err != nil {
-				return &scheduler.ErrFailedToApplyCustomSpec{
-					Name:  obj.Name,
-					Cause: fmt.Sprintf("Failed to apply spec: %v. Err: %v", obj.Name, err),
-					Type:  obj,
-				}
+	if obj, ok := specObj.(*stork_api.ClusterPair); ok {
+		logrus.Info("Applying clusterpair spec")
+		err = k8sOps.CreateClusterPair(obj)
+		if err != nil {
+			return nil, &scheduler.ErrFailedToApplyCustomSpec{
+				Name:  obj.Name,
+				Cause: fmt.Sprintf("Failed to apply spec: %v. Err: %v", obj.Name, err),
+				Type:  obj,
 			}
-			if err := k8sOps.ValidateClusterPair(obj.Name, defaultTimeout, defaultRetryInterval); err != nil {
-				return &scheduler.ErrFailedToApplyCustomSpec{
-					Name:  obj.Name,
-					Cause: fmt.Sprintf("Failed to validate cluster Pair: %v. Err: %v", obj.Name, err),
-				}
+		}
+		if err := k8sOps.ValidateClusterPair(obj.Name, defaultTimeout, defaultRetryInterval); err != nil {
```

Comment:
This function should only create the object, the validation should be done in WaitForRunning()

Reply:
Moved